src/scraper/scraper.cpp

```diff
@@ -5243,104 +5240,73 @@ scraperSBvalidationtype ValidateSuperblock(const NN::Superblock& NewFormatSuperb
+                        // Notice the below is NOT using the time. We switch to the content only. The time is only used to make sure
+                        // we test the convergence of the project objects in time order, but once a content hash is selected based on the time,
+                        // only the content hash is used to count occurrences in the multimap, because the times for the same
+                        // project object (part hash) will be different across different manifests and different scrapers.
+                        unsigned int nIdenticalContentManifestCount = mProjectObjectsBinnedbyContent.count(std::get<0>(iter.second));
+                        if (nIdenticalContentManifestCount >= NumScrapersForSupermajority(nScraperCount))
                         {
-                            // ------------------------------------------ full content hash ---- project name
-                            mMatchingProjectPartHashes.insert(std::make_pair(iter.first, iWhitelistProject.m_name));
-                            bMatched = true;
-                        }
-                    }
-                }
-
-                // Keep track of the number of matched projects.
-                if (bMatched) ++iCountMatchedProjects;
-
-                if (fDebug3) _log(logattribute::INFO, "ENDLOCK", "CScraperManifest::cs_mapManifest");
-            }
-
-            uint256 nLatestProjectPartMatchingHash;
-            int64_t nLatestProjectPartMatchingTime = 0;
-            uint256 nLatestProjectPartMatchingConsensusBlock;
-            uint256 nLatestProjectPartMatchingManifestForBeaconList;
-
-            for (const auto& iter : mMatchingProjectPartHashes)
-            {
-                // We want the most recent (latest) part, constrained to the above map. This is slightly different from.
-                // but similar to the construction of the convergence without constraint.
-                for (const auto& iter_inner : mProjectObjectsBinnedByTime)
-                {
-                    // Only go until the first match (latest), then break.
-                    if (std::get<0>(iter_inner.second) == iter.first)
-                    {
-                        nLatestProjectPartMatchingHash = iter.first;
-                        nLatestProjectPartMatchingTime = iter_inner.first;
-                        nLatestProjectPartMatchingConsensusBlock = get<1>(iter_inner.second);
-                        nLatestProjectPartMatchingManifestForBeaconList = get<2>(iter_inner.second);
-
-                        break;
-                    }
-                }
-            }
-
-            unsigned int nIdenticalContentManifestCount = mProjectObjectsBinnedbyContent.count(nLatestProjectPartMatchingHash);
+                            // Get the actual part ----------------- by object hash.
+                            auto iPart = CSplitBlob::mapParts.find(std::get<0>(iter.second));
```

Comment:
For the sake of the reference implementation, we may want a lock on `cs_mapParts` for this too.

Reply:
I think you are right.